kong/api/routes/upstreams.lua

```diff
@@ -167,17 +167,18 @@ return {
     end
   },
 
-  ["/upstreams/:name_or_id/targets/:target"] = {
+  ["/upstreams/:name_or_id/targets/:target_or_id"] = {
     before = function(self, dao_factory, helpers)
       crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      crud.find_target_by_target_or_id(self, dao_factory, helpers)
     end,
 
     DELETE = function(self, dao_factory)
       clean_history(self.upstream.id, dao_factory)
 
       -- this is just a wrapper around POSTing a new target with weight=0
       local data, err = dao_factory.targets:insert({
-        target      = self.params.target,
+        target      = self.target.target,
```

Comment:
The `target` field name is confusing with the `target` entity, should be really called `address` ("Targets can have an address and a weight" as opposed to "Targets can have a target and a weight")